tex/gregoriotex.lua

```diff
@@ -120,6 +120,37 @@ local translation_mark = 1
 local abovelinestext_mark = 2
 log("marker whatsit id is %d", marker_whatsit_id)
 
+local function gregorio_exe()
+  if real_gregorio_exe == nil then
+    local exe_version
+
+    -- first look for one with the exact version
+    real_gregorio_exe = 'gregorio-5_0_0-rc1' -- FILENAME_VERSION
+    exe_version = io.popen(real_gregorio_exe..' --version', 'r')
+    exe_version = exe_version:read("*line")
+    if not exe_version then
+      -- look for suffix-less executable
+      real_gregorio_exe = 'gregorio'
+      exe_version = io.popen(real_gregorio_exe..' --version', 'r')
+      exe_version = exe_version:read("*line")
+      if not exe_version or string.match(exe_version,"%d+%.%d+%.")
+          ~= string.match(internalversion,"%d+%.%d+%.") then
+        real_gregorio_exe = nil
```

Comment:
Okay, I think I've figured the sequence now, and this line throws a flag for me.  The basic process is "if I don't know what the name of the executable is (i.e. `real_gregorio_exe` is `nil`) then figure out the name and proceed".  `real_gregorio_exe` is initialized to nil, so this block will run the first time.  After that, we know the name and simply proceed.  Thus things work fine when we can find the executable.  However, if we can't find the executable, then we set `real_gregorio_exe` back to `nil` which means that we're going to try again when we need it again (which of course will also fail).  As a result, we're going to try to find the executable (and fail) each time we need it.  Once we've failed once, we don't really need to keep checking for the executable.